backend/tomato/elements/generic.py

```diff
@@ -238,6 +239,46 @@ def after_upload_use(self):
 		self.custom_template = True
 		self.save()
 
+	def checkMigrate(self):
+		#We only migrate if the element is prepared
+		return self.state in [ST_PREPARED]
+	
+	def action_migrate(self,host):
+		
+		from ..host import Host
+		
+		host = Host.objects.filter(name = host)
+		
+		UserError.check(host, code=UserError.NO_RESOURCES, message="Host not found",data={"type": self.TYPE})
```

Comment:
You should first try to actually find a host when no host is given in the API call